internal/helpers/rollout_restart.go

```diff
@@ -135,6 +140,14 @@ func patchForRolloutRestart(ctx context.Context, obj ctrlclient.Object, client c
 		}
 		t.Spec.Template.ObjectMeta.Annotations[AnnotationRestartedAt] = time.Now().Format(time.RFC3339)
 		return client.Patch(ctx, t, patch)
+
+	case *rolloutsv1alpha1.Rollout:
+		patch := ctrlclient.MergeFrom(t.DeepCopy())
```

Comment:
VSO will need to have the proper RBAC permissions to patch this new resource type. We will need to extend the RBAC role to include it. I think you can probably add:
```
//+kubebuilder:rbac:groups=argoproj.io,resources=rollouts,verbs=get;list;watch;patch
```

To each controller that handles secret sync. E.g: https://github.com/hashicorp/vault-secrets-operator/blob/648d116d0b8f2ef8a1a99a11a5c17fd1a0837f88/controllers/vaultdynamicsecret_controller.go#L66

We will need to update:
```
controllers/hcpvaultsecretsapp_controller.go
controllers/vaultdynamicsecret_controller.go
controllers/vaultpkisecret_controller.go
controllers/vaultstaticsecret_controller.go
```

Reply:
Looks like this does not apply for `controllers/hcpvaultsecretsapp_controller.go` please correct me if I am missing something.

Reply:
Updated also for `controllers/hcpvaultsecretsapp_controller.go`. Please let me know if you anymore feedback.